colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +210,72 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+def _parse_cmake_version():
+    """
+    Parse the CMake version number by running `cmake --version`.
+
+    The version number is parse from the output of 'cmake --version' and is
+    expected in the form 'cmake version #.#.#' where each '#' character
+    represents part of the version number. Additional text may follow, however
+    only the first line is parsed. Parsing is performed by the pkg_resources
+    package and the returned object is a Version object from that package.
+
+    This function blocks on the execution of cmake and should only be used to
+    cache the CMake version number. External API users should use
+    get_cmake_version()
+
+    :returns: The version as parsed by the pkg_resources package
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    try:
+        output = subprocess.check_output([CMAKE_EXECUTABLE, '--version'])
+        lines = output.decode().splitlines()
+        ver_line = lines[0] if lines and len(lines) else None
```

Comment:
The condition `len(lines)` is redundant with `lines`.

The logic can also be reduced by removing this line, changing the condition in the next line to just `if lines:`, and then below use `lines[0]` instead of `ver_line`.

Reply:
Fixed